### PR TITLE
[RDBMS] fix help statements for `az postgres flexible-server virtual-endpoint` and `az postgres flexible-server replica stop-replication`

### PR DESCRIPTION
**Pull request by nasc17**
**Related command**
`az postgres flexible-server virtual-endpoint`
`az postgres flexible-server replica stop-replication`

**Description**
For virtual-endpoint commands: the virtual endpoint name validation does not allow '_' but help text gives example name as 'test_virtual_endpoint'. Edited help text to include hyphen instead?

For stop-replication: this command name has been deprecated. Users should call promote to perform the desired operation. Edited help text to explain this.

**Testing Guide**
Verified CLI changes manually.
image
image

**History Notes**
[RDBMS] `az postgres flexible-server replica stop-replication`: Stop replication to a read replica and make it a read/write server. This command is deprecated. Use `az postgres flexible-server replica promote` instead.

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Review by rmendiza — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/_helptext_pg.py [hunk 1]
```diff
@@ -759,7 +759,7 @@
 
 helps['postgres flexible-server replica stop-replication'] = """
 type: command
-short-summary: Stop replication to a read replica and make it a read/write server.
+short-summary: Stop replication to a read replica and make it a read/write server. This command is deprecated. Use 'az postgres flexible-server replica promote' instead.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_helptext_pg.py:762]: If you want to deprecate a command, pls modify the code in command definition. FYI https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

**Comment by nasc17** [comment]
@evelyn-ys the deprecation commands have been modified. Could you please review, thank you

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 2]
```diff
@@ -829,9 +829,6 @@ def _flexible_server_params(command_group):
                 c.argument('public_access', options_list=['--public-access'], arg_type=get_enum_type(['Enabled', 'Disabled']),
                            help='Determines the public access. ')
 
-        with self.argument_context('{} flexible-server replica stop-replication'.format(command_group)) as c:
-            c.argument('server_name', arg_type=server_name_arg_type)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:833]: Why remove this parameter definition?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py [hunk 3]
```diff
@@ -212,7 +212,7 @@ def load_flexibleserver_command_table(self, _):
                             custom_command_type=flexible_servers_custom_postgres,
                             client_factory=cf_postgres_flexible_servers) as g:
         g.custom_command('create', 'flexible_replica_create', supports_no_wait=True)
-        g.custom_command('stop-replication', 'flexible_replica_stop', confirmation=True)
+        g.custom_command('stop-replication', 'flexible_replica_stop', confirmation=True, deprecate_info=g.deprecate(target='stop-replication', redirect='promote', hide=True))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_commands.py:215]:
```suggestion
        g.custom_command('stop-replication', 'flexible_replica_stop', confirmation=True, deprecate_info=g.deprecate(redirect='postgres flexible-server replica promote', hide=True))
```

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (nasc17) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:833]: Redundant, is already stated to be needed for all postgres flexible-server commands at line [547](https://github.com/Azure/azure-cli/blob/3b47eed45a24512fe5494555fe28f32f29a1e870/src/azure-cli/azure/cli/command_modules/rdbms/_params.py#L547)
image

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**